MembraneRTC/src/main/java/org/membraneframework/rtc/media/SoundDetection.kt

```diff
@@ -0,0 +1,193 @@
+     *
+     * @param monitorInterval The time interval (in milliseconds) between sound detection checks.
+     * @param samplingRate The audio sampling rate in Hz.
+     * @param volumeThreshold The threshold value in decibels (dB) above which a sound is considered detected.
+     */
+    @SuppressLint("MissingPermission")
+    fun start(monitorInterval: Int = 1, samplingRate: Int = 22050, volumeThreshold: Int = -100) {
+        if (isRecording) {
+            Timber.w("Sound detection is already in progress. Ignoring the start request.")
+            return
+        }
+
+        bufferSize = AudioRecord.getMinBufferSize(
+            samplingRate,
+            AudioFormat.CHANNEL_IN_MONO,
+            AudioFormat.ENCODING_PCM_16BIT
+        )
+        audioRecord = try {
+            AudioRecord(
+                MediaRecorder.AudioSource.MIC,
+                samplingRate,
+                AudioFormat.CHANNEL_IN_MONO,
+                AudioFormat.ENCODING_PCM_16BIT,
+                bufferSize
+            )
+        } catch (e: SecurityException) {
+            throw SecurityException(
+                "Unable to initialize the AudioRecord." +
+                    " Ensure that the recording permission is granted."
+            )
+        }
+
+        if (audioRecord?.state == AudioRecord.STATE_INITIALIZED) {
+            audioRecord?.startRecording()
+            isRecording = true
+            startTimer(monitorInterval, volumeThreshold)
+        } else {
+            Timber.e("COULDNT_PREPARE_RECORDING AudioRecord couldn't be initialized.")
+            throw RuntimeException("AudioRecord couldn't be initialized.")
+        }
+    }
+
+    /**
+     * Stops the sound detection process and releases the resources used by the AudioRecord.
+     */
+    fun stop() {
+        if (!isRecording) {
+            Timber.e("INVALID_STATE Please call start before stopping recording")
+            return
+        }
+        stopTimer()
+        isRecording = false
+        try {
+            audioRecord?.stop()
+            audioRecord?.release()
+        } catch (e: RuntimeException) {
+            Timber.e(
+                "RUNTIME_EXCEPTION No valid audio data received. You may be using a device that can't record audio."
+            )
+        } finally {
+            audioRecord = null
+        }
+    }
+
+    /**
+     * Sets a listener to receive sound detection events.
+     *
+     * @param listener The listener to be notified when a sound is detected.
+     */
+    fun setSoundDetectionListener(listener: OnSoundDetectedListener?) {
+        onSoundDetectedListener = listener
+    }
+
+    /**
+     * Sets the sound detection status and notifies the listener if available.
+     *
+     * @param detectionResult The result of the sound detection (true if sound detected, false otherwise).
+     */
+    private fun setIsSoundDetected(detectionResult: Boolean) {
```

Comment:
In case of private functions you don't need to document so much. We mostly care for public, user-facing API (users shouldn't need to look at our code).

For private apis is mostly quicker and easier to maintain to just look at the code, especially if functions are well named (it's obvious that `readAudioData` reads audio data, no need to comment that). In time, comments like this can become outdated. Of course, we should still comment complicated, hacky or non-obcious code (like the formula for sound loudness)